src/api/spec/controllers/search_controller_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe SearchController, vcr: true do
+  render_views
+
+  shared_examples 'block anonymous request' do
+    it { expect(response).to have_http_status(:redirect) }
+  end
+
+  shared_examples 'find project' do
+    it { expect(response).to have_http_status(:success) }
+    it { expect(Xmlhash.parse(response.body)).to have_key('project') }
+    it { expect(Xmlhash.parse(response.body)['project']).to have_key('name') }
+    it { expect(Xmlhash.parse(response.body)['project']['name']).to eq(user.home_project.name) }
+  end
+
+  shared_examples 'find package' do
+    it { expect(response).to have_http_status(:success) }
+    it { expect(Xmlhash.parse(response.body)).to have_key('package') }
+    it { expect(Xmlhash.parse(response.body)['package']).to have_key('name') }
+    it { expect(Xmlhash.parse(response.body)['package']['name']).to eq('apacheX') }
+    it { expect(Xmlhash.parse(response.body)['package']['project']).to eq(user.home_project.name) }
+  end
+
+  describe 'search for projects' do
+    context 'unauthenticated' do
+      subject! { get :project, params: { match: "@name='foobar'" } }
+
+      it_behaves_like 'block anonymous request'
+    end
+
+    context 'authenticated' do
+      let!(:user) { create(:confirmed_user, :with_home, login: 'foo') }
+
+      before do
+        login user
+      end
```

Comment:
You can move this into the shared example.

Reply:
done